dev_docs/tutorials/setting_up_a_development_env.mdx

```diff
@@ -0,0 +1,89 @@
+---
+id: kibDevTutorialSetupDevEnv
+slug: /kibana-dev-docs/tutorial/setup-dev-env
+title: Setting up a Development Environment
+summary: Learn how to setup a development environemnt for contributing to the Kibana repository
+date: 2021-04-26
+tags: ['kibana', 'onboarding', 'dev', 'architecture', 'setup']
+---
+
+Setting up a development environment is pretty easy.
+
+<DocCallOut title="A note about Windows">
+  In order to support Windows development we currently require you to use one of the following:
+
+  - [Git Bash](https://git-scm.com/download/win)
+  - [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/about)
+
+
+  Before running the steps below, please make sure you have installed [Visual C++ Redistributable for Visual Studio 2015](https://www.microsoft.com/en-us/download/details.aspx?id=48145) and that you are running all commands in either Git Bash or WSL.
+</DocCallOut>
+
+## Get the code
+
+Start by forking [the Kibana repository](https://github.com/elastic/kibana) on Github so that you have a place to stage pull requests and create branches for development.
+
+Then clone the repository to your machine:
+
+```sh
+git clone https://github.com/[YOUR_USERNAME]/kibana.git kibana
+cd kibana
+```
+
+## Install dependencies
+
+Install the version of Node.js listed in the `.node-version` file. This can be automated with tools such as [nvm](https://github.com/creationix/nvm) or [nvm-windows](https://github.com/coreybutler/nvm-windows). As we also include a .nvmrc file you can switch to the correct version when using nvm by running:
```

Comment:
```suggestion
Install the version of Node.js listed in the `.node-version` file. This can be automated with tools such as [nvm](https://github.com/creationix/nvm) or [nvm-windows](https://github.com/coreybutler/nvm-windows). As we also include a `.nvmrc` file you can switch to the correct version when using nvm by running:
```